### PR TITLE
Keep tunnel open

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Adds logic to keep tunnel open until exploited machines connect. This preserves the existing logic and makes no attempts to improve upon it. 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running locally from source
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class ExploitInterceptingTelemetryMessenger(ITelemetryMessenger):
+    def __init__(self, telemetry_messenger: ITelemetryMessenger, tunnel: MonkeyTunnel):
+        self._telemetry_messenger = telemetry_messenger
+        self._tunnel = tunnel
+
+    def send_telemetry(self, telemetry: ITelem):
+        if (
+            telemetry.telem_category == TelemCategoryEnum.EXPLOIT
+            and telemetry.propagation_result is True
```
Comment (VakarisZ) [inline monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py:15]: There's no `propagation_result` in `ITelem`

File: monkey/infection_monkey/tunnel.py [hunk 2]
```diff
@@ -195,5 +204,17 @@ def get_tunnel_for_ip(self, ip: str):
         ip_match = get_interface_to_target(ip)
         return "%s:%d" % (ip_match, self.local_port)
 
+    def report_exploited_machine(self):
```
Comment (VakarisZ) [inline monkey/infection_monkey/tunnel.py:207]: this method doesn't report anything. I'd say a better name would be `on_successful_exploitation` to indicate that it needs to be called when machine got successfully exploited?

File: monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py:15]: Yeah, but we've already verified that it's an ExploitTelem. How do we work around this so the type checker is happy. Can we "cast" it somehow?

Comment (mssalvatore) [inline monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py:15]:
```suggestion
            and telemetry.propagation_result is True  # type: ignore
```

File: monkey/infection_monkey/tunnel.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/infection_monkey/tunnel.py:207]: `increment_exploited_machine_counter()`?
`mark_machines_exploited()`?
`set_machine_successfully_exploited()`?

File: monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py:15]: Maybe I can avoid the issue with singledispatch.

Comment (mssalvatore) [reply monkey/infection_monkey/telemetry/messengers/exploit_intercepting_telemetry_messenger.py]: I ended up using singledispatch.

**Pull request merged**